common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -110,15 +123,16 @@ def test_page_view_event(self):
         And I visit my account settings page
         Then a page view analytics event should be recorded
         """
-        self.verify_events_of_type(
-            self.username,
-            u"edx.user.settings.viewed",
-            [{
-                u"user_id": long(self.user_id),
-                u"page": u"account",
-                u"visibility": None,
-            }]
-        )
+
+        event_filter = {
+            'event_type':'edx.user.settings.viewed',
+            'event': {
+                'user_id': self.user_id,
+                'page': 'account',
+                'visibility': None
+            }
+        }
+        self.wait_for_events(event_filter=event_filter, number_of_matches=1)
```

Comment:
I'm thinking through this as well, and I see @andy-armstrong's point. (Though...not ready to fully agree...). However, just to think on this some more, I could see some organization like this:
* a boolean on whether or not an event is found (strict or not strict)
* a plain old assertEqual (or similar) that tells you if the event was found or not
* a wait_for, that would wait for the boolean to return True

That might more closely match bok-choy's approach with page objects.